packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -46,49 +48,16 @@ function isAncestorPath(ancestor: string, descendant: string): boolean {
   return !path.relative(ancestor, descendant).startsWith('..');
 }
 
-export type BatchedTelemetry = {
-  telemetryName: string;
-  scanResults: ScanResults;
-  msElapsed: number;
-  msInserted: number;
-};
-
 export class Watcher {
   config?: TimestampedConfiguration;
   appmapWatcher?: chokidar.FSWatcher;
   appmapPoller?: chokidar.FSWatcher;
   configWatcher?: chokidar.FSWatcher;
-  batchedTelemetry: BatchedTelemetry[] = [];
-  sendPeriodMilliseconds = 5000;
-  interval: NodeJS.Timer;
+  telemetry: WatchScanTelemetry | undefined;
+  scanEventEmitter = new EventEmitter();
 
   constructor(private options: WatchScanOptions) {
-    this.interval = setInterval(() => {
-      this.sendScanResultsTelemetryInABatch();
-    }, this.sendPeriodMilliseconds);
-    // must unref because there's no condition under which the timer
-    // will stop; else the process doesn't exit
-    this.interval.unref();
-  }
-
-  public sendScanResultsTelemetryInABatch() {
-    const dateNow = Date.now();
-    if (
-      this.batchedTelemetry &&
-      this.batchedTelemetry.length > 0 &&
-      dateNow - this.batchedTelemetry[this.batchedTelemetry.length - 1].msInserted >
-        this.sendPeriodMilliseconds
-    ) {
-      const aggregatedScanResults: ScanResults = new ScanResults();
-      let totalMsElapsed = 0;
-      this.batchedTelemetry.forEach((item) => {
-        aggregatedScanResults.aggregate(item.scanResults);
-        totalMsElapsed += item.msElapsed;
-      });
-
-      sendScanResultsTelemetry(aggregatedScanResults, totalMsElapsed);
-      this.batchedTelemetry = [];
-    }
+    if (options.sendTelemetry) this.telemetry = new WatchScanTelemetry(this.scanEventEmitter);
```

Comment:
Not sure if this option is really needed. Telemetry can be easily stubbed in tests.